File: B_Vlad_and_Shapes.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int t;
    cin >> t;  // number of test cases

    while (t--) {
        int n;
        cin >> n;  // size of the grid

        vector<string> grid(n);
        for (int i = 0; i < n; i++) {
            cin >> grid[i];
        }

        // solution
        vector<pair<int, int> > arr;
        for (int i = 0; i < n; i++) {
            int l = 0;
            int r = grid[0].size() - 1;
            int ct = 0;
            while (l < r && ct <= 2) {
                if (grid[i][l] != '1') {
                    l++;
                }
                if (grid[i][r] != '1') {
                    r--;
                }
                if (grid[i][l] == '1' && grid[i][r] == '1') {
                    arr.push_back(make_pair(l, r));
                    ct++;
                    break;
                }
            }
        }

        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] == arr[i + 1]) {
                cout << "SQUARE" << endl;
            } else {
                cout << "TRIANGLE" << endl;
            }
            break;
        }
    }

    return 0;
}

File: B_Vlad_and_Shapes.py
t = int(input())  # number of test cases

for _ in range(t):
    n = int(input())  # size of the grid
    grid = []
    arr = []
    for _ in range(n):
        row = input().strip()
        grid.append(row)

    # solution
    i = 0
    while i < n:
        l = 0
        r = len(grid[0])-1
        ct = 0
        while l < r and ct <= 2:
            if grid[i][l] != "1":
                l += 1
            if grid[i][r] != "1":
                r -= 1
            if grid[i][l] == "1" and grid[i][r] == "1":
                arr.append((l, r))
                ct += 1
                break
        i += 1
    for ele in range(len(arr)): 
        if arr[ele] == arr[ele+1]:
            print("SQUARE")
        else:
            print("TRIANGLE")
        break

